Sources/Vapor/Utilities/FileIO.swift

```diff
@@ -93,20 +94,21 @@ public struct FileIO: Sendable {
         chunkSize: Int = NonBlockingFileIO.defaultChunkSize,
         onRead: @Sendable @escaping (ByteBuffer) -> EventLoopFuture<Void>
     ) -> EventLoopFuture<Void> {
-        guard
-            let attributes = try? FileManager.default.attributesOfItem(atPath: path),
-            let fileSize = attributes[.size] as? NSNumber
-        else {
-            return self.request.eventLoop.makeFailedFuture(Abort(.internalServerError))
+        return self.request.eventLoop.makeFutureWithTask { () -> Int64? in
+            let fileInfo = try await FileSystem.shared.info(forFileAt: .init(path))
+            return fileInfo?.size
+        }.flatMap { fileSize in
+            guard let fileSize = fileSize else {
+                return self.request.eventLoop.makeFailedFuture(Abort(.internalServerError))
+            }
+            return self.read(
+                path: path,
+                fromOffset: 0,
+                byteCount: Int(fileSize),
+                chunkSize: chunkSize,
+                onRead: onRead
+            )
```

Comment:
```suggestion
        self.request.eventLoop.makeFutureWithTask {
            guard let fileSize = try await FileSystem.shared.info(forFileAt: .init(path))?.size else {
                throw Abort(.internalServerError)
            }
            try await self.read(
                path: path,
                fromOffset: 0,
                byteCount: Int(fileSize),
                chunkSize: chunkSize,
                onRead: onRead
            ).get()
        }
```
Saves at least one extra thread hop, especially in the case of the file not existing (where it saves more like two or three).